db_refactor/instance_schema.sql

```diff
@@ -0,0 +1,733 @@
+  scopes text[]
+);
+
+DROP TABLE IF EXISTS zitadel.idp_ldap CASCADE;
+CREATE TABLE zitadel.idp_ldap (
+  idp_provider_id INTEGER NOT NULL,
+  CONSTRAINT idp_provider_id_fk FOREIGN KEY(idp_provider_id) REFERENCES idp_providers(id),
+  -- TODO check ALL sizes
+  start_tls BOOLEAN NOT NULL,
+  base_dn VARCHAR(100) NOT NULL,
+  bind_dn VARCHAR(100)  NOT NULL,
+  bind_password VARCHAR(100) NOT NULL,
+  user_base VARCHAR(100) NOT NULL,
+  timeout BIGINT NOT NULL,
+  id_attribute VARCHAR(50),
+  first_name_attribute VARCHAR(50),
+  last_name_attribute VARCHAR(50),
+  display_name_attribute VARCHAR(100),
+  nick_name_attribute  VARCHAR(100),
+  preferred_username_attribute VARCHAR(100),
+  email_attribute VARCHAR(100),
+  email_verified VARCHAR(100),
+  phone_attribute VARCHAR(100),
+  phone_verified_attribute VARCHAR(100),
+  preferred_language_attribute VARCHAR(100),
+  avatar_url_attribute VARCHAR(100),
+  profile_attribute VARCHAR(100),
+  user_object_classes text[] NOT NULL,
+  user_filters text[] NOT NULL,
+  servers text[] NOT NULL,
+  root_ca bytea
+);
+
+DROP TABLE IF EXISTS zitadel.idp_google CASCADE;
+CREATE TABLE zitadel.idp_google (
+  idp_provider_id INTEGER NOT NULL,
+  CONSTRAINT idp_provider_id_fk FOREIGN KEY(idp_provider_id) REFERENCES idp_providers(id),
+  client_id VARCHAR(100) NOT NULL,
+  client_secret JSONB  NOT NULL,
+  scopes text[]
+);
+
+DROP TABLE IF EXISTS zitadel.idp_gitlab_self_hosted CASCADE;
+CREATE TABLE zitadel.idp_gitlab_self_hosted (
+  idp_provider_id INTEGER NOT NULL,
+  CONSTRAINT idp_provider_id_fk FOREIGN KEY(idp_provider_id) REFERENCES idp_providers(id),
+  issuer VARCHAR(200) NOT NULL,
+  client_id VARCHAR(100) NOT NULL,
+  client_secret JSONB  NOT NULL,
+  scopes text[]
+);
+
+DROP TABLE IF EXISTS zitadel.idp_gitlab CASCADE;
+CREATE TABLE zitadel.idp_gitlab (
+  idp_provider_id INTEGER NOT NULL,
+  CONSTRAINT idp_provider_id_fk FOREIGN KEY(idp_provider_id) REFERENCES idp_providers(id),
+  client_id VARCHAR(100) NOT NULL,
+  client_secret JSONB NOT NULL,
+  -- TODO check if there's a better type for scopes than text[]
+  scopes text[]
+);
+
+DROP TABLE IF EXISTS zitadel.idp_github_enterprise CASCADE;
+CREATE TABLE zitadel.idp_github_enterprise (
+  idp_provider_id INTEGER NOT NULL,
+  CONSTRAINT idp_provider_id_fk FOREIGN KEY(idp_provider_id) REFERENCES idp_providers(id),
+  client_id VARCHAR(100) NOT NULL,
+  client_secret JSONB  NOT NULL,
+  authorization_endpoint VARCHAR(200) NOT NULL,
+  token_endpoint VARCHAR(200) NOT NULL,
+  user_endpoint VARCHAR(100) NOT NULL,
+  scopes text[]
+);
+
+DROP TABLE IF EXISTS zitadel.idp_github CASCADE;
+CREATE TABLE zitadel.idp_github(
+  idp_provider_id INTEGER NOT NULL,
+  CONSTRAINT idp_provider_id_fk FOREIGN KEY(idp_provider_id) REFERENCES idp_providers(id),
+  client_id VARCHAR(100) NOT NULL,
+  client_secret JSONB NOT NULL,
+  scopes text[]
+);
+
+DROP TYPE IF EXISTS zitadel.tenant CASCADE;
+CREATE TYPE zitadel.tenant AS ENUM ('consumer', 'common', 'organization');
```

Comment:
i would clarify that this enum is for azure so i would call it zitadel.azure_tenant_type